erasure-healfile.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * Minio Cloud Storage, (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import "encoding/hex"
+
+// erasureHealFile heals part file of an object. It never writes the healed file to the actual
+// location, but to a temporary location ".minio/tmpPath" as specified by the caller. It will
+// be the responsibility of the caller to rename to the actual location.
+func erasureHealFile(latestDisks []StorageAPI, outDatedDisks []StorageAPI,
+	volume, // Volume of the object.
+	path, // Path to the part file.
+	partName, // Name of the part.
+	metaBucket, // .minio
+	tmpPath string, // Temporary file path where healed part will be written to.
+	size int64, // Size of the part file.
+	erasure erasureInfo) ([]checkSumInfo, error) {
+	var offset int64
+	remainingSize := size
+
+	// Sha512 hash for bitrot protection.
```

Comment:
'// Sha512 hash for bitrot protection' -> '// Sha512 hash for bit rot protection'